.github/workflows/build.yaml

```diff
@@ -0,0 +1,40 @@
+name: Build
+on: [push, pull_request]
+jobs:
+  build_docker:
+    name: Build Docker Image
+    runs-on: ubuntu-20.04
+    env:
+      DOCKER_TAG: cabforum/build-guidelines-action
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Build Image
+        run: docker build --tag ${{ env.DOCKER_TAG}} .
+      - name: Save image
+        run: |
+          docker save ${{ env.DOCKER_TAG }} |\
+          gzip > build-guidelines-action.tgz
+      - name: Upload archive
+        uses: actions/upload-artifact@v1
+        with:
+          name: build-guidelines-action
+          path: build-guidelines-action.tgz
+  test_docker:
+    name: Test Docker Image
+    runs-on: ubuntu-20.04
+    needs:
+      - build_docker
+    env:
+      DOCKER_TAG: cabforum/build-guidelines-action
+    steps:
+      - name: Fetch Image
+        uses: actions/download-artifact@v1
+        with:
+          name: build-guidelines-action
```

Comment:
I'd also specify a `path:` here, so that the `docker load` on line 36 doesn't have to be "guessing" at the path.